esp/src/src/Memory.ts

```diff
@@ -0,0 +1,97 @@
+import * as QueryResults from "dojo/store/util/QueryResults";
+import * as SimpleQueryEngine from "dojo/store/util/SimpleQueryEngine";
+import { alphanumSort } from "./Utility";
+
+//  Replacement for "dojo/store/Memory"
+export interface MemoryOptions {
+    data?: any,
+    idProperty?: string,
+    index?: any,
+    queryEngine?: any
+}
+
+export class Memory {
+
+    data = null;
+    idProperty = "id";
+    index = null;
+    queryEngine = SimpleQueryEngine;
+
+    constructor(options: MemoryOptions = {}) {
+        for (const i in options) {
+            this[i] = options[i];
+        }
+        this.setData(this.data || []);
+    }
+
+    get(id) {
+        return this.data[this.index[id]];
+    }
+
+    getIdentity(object) {
+        return object[this.idProperty];
+    }
+
+    put(row, options) {
+        const data = this.data;
+        const index = this.index;
+        const idProperty = this.idProperty;
+        const id = row[idProperty] = (options && "id" in options) ? options.id : idProperty in row ? row[idProperty] : Math.random();
```

Comment:
It would be rare but `Math.random()` could result in duplicate id.

Reply:
True - hope I do the lotto on that day!

FYI - This is ported from dojo/store/Memory "as is" and I am pretty sure we always use a valid ID so it never comes into play.